src/sqlfluff/rules/L069.py

```diff
@@ -0,0 +1,175 @@
+                                    "comma",
+                                )
+                            ),
+                        )
+                    )
+                )
+
+                # we only care about 2-arguments convert
+                # some dialects allow an optional 3rd argument e.g TSQL
+                # which cannot be rewritten into CAST
+                if len(convert_content) > 2:
+                    return None
+
+                # get the expression
+                expression = convert_content[1].raw
+
+                # get the data type
+                datatype = convert_content[0].raw
+
+                # Create fix to replace ``CONVERT`` with ``CAST``.
+                fix = LintFix.replace(
+                    context.segment,
+                    [
+                        CodeSegment(
+                            raw=f"CAST({expression} AS {datatype})",
+                        )
+                    ],
+                )
+
+                return LintResult(
+                    anchor=context.segment,
+                    fixes=[fix],
+                    description=f"Use 'CAST' instead of \n"
+                    f"'{context.segment.get_child('function_name').raw_upper}'.",
+                )
+
+            # casting_operator(::)
+            if context.segment.is_type("cast_expression"):
+                # get the expression and the datatype segment
+                expression_datatype_segment = FunctionalContext(
+                    context
+                ).segment.children(
+                    sp.and_(
+                        sp.not_(sp.is_meta()),
+                        sp.not_(
+                            sp.is_type(
+                                "start_bracket",
+                                "end_bracket",
+                                "whitespace",
+                                "newline",
+                                "casting_operator",
+                            )
+                        ),
+                    )
+                )
+
+                # we can have multicast e.g 1::int::text
+                # in that case, we need to introduce multiple CAST()
+                # start off new_segment as the expression
+                new_segment = expression_datatype_segment[0].raw
+
+                for datatype_level in range(len(expression_datatype_segment[1:])):
+                    # skip the first element(the expression)
+                    datatype_level = datatype_level + 1
+                    # update new_segment to include datatype
+                    # as well as accommodate multicast
+                    new_datatype = expression_datatype_segment[datatype_level].raw
+                    new_segment = f"CAST({new_segment} AS {new_datatype})"
+
+                # create fix to replace ``::`` with ``CAST``.
+                fix = LintFix.replace(
+                    context.segment,
+                    [
+                        CodeSegment(
```

Comment:
Same comment as before: can you create individual segments?

Reply:
Fixed by [ebeb4ad](https://github.com/sqlfluff/sqlfluff/pull/3747/commits/ebeb4ada735810c3d050d690d93e0d8fcc64d4fa)